### PR TITLE
generalize iNS manufactured solution for Hdiv

### PR DESCRIPTION
**Pull request by richardschu**
introduces periodic boundary conditions for the manufactured solution case s.t. we can check the accuracy of Hdiv. Was this done anywher? 
I will also use it to test restart.

@kronbichler this will be the testbed to check the restart mechanics properly (will merge selected commits to the other branch).

### CODE REVIEW

File: applications/incompressible_navier_stokes/navier_stokes_manufactured/application.h [hunk 1]
```diff
@@ -605,28 +614,51 @@ class Application : public ApplicationBase<dim, Number>
...
+      // Save the *coarse* triangulation for later deserialization.
+      if(write_restart and this->param.grid.triangulation_type == TriangulationType::Serial)
+      {
+        save_coarse_triangulation<dim>(this->param.restart_data, tria);
+      }
 
-        if(global_refinements > 0)
-          tria.refine_global(global_refinements);
-      };
+      if(global_refinements > 0)
```
Comment (richardschu) [inline applications/incompressible_navier_stokes/navier_stokes_manufactured/application.h]: we refine afterwards, since we have to store the coarse tria in case of restart. I want to write some tests for restart next.

File: include/exadg/incompressible_navier_stokes/spatial_discretization/spatial_operator_base.cpp [hunk 2]
```diff
@@ -171,11 +171,6 @@ SpatialOperatorBase<dim, Number>::initialize_dof_handler_and_constraints()
   }
   print_parameter(pcout, "number of dofs per cell", fe_u->n_dofs_per_cell());
   print_parameter(pcout, "number of dofs (total)", dof_handler_u.n_dofs());
-  if(param.spatial_discretization == SpatialDiscretization::HDIV)
```
Comment (richardschu) [inline include/exadg/incompressible_navier_stokes/spatial_discretization/spatial_operator_base.cpp]: are the DoFs that are periodic not counted for the L2 case? I think this message would apply in any case, and hence does not really help. Since we are using DG, the periodic BCs only tell the boundary cells, which cells are their neighbors on the other side of the domain, right? So there are never any DoFs reduced ..?
